lib/spack/spack/bootstrap/core.py

```diff
@@ -468,6 +479,66 @@ def _add_externals_if_missing() -> None:
     )
 
 
+class BootstrapResource:
+    """Represents a resource required by Spack to run, fetched as part
+    of the bootstrapping operation
+
+    More or less identical to package based resources but exposes additional logic related
+    to bootstrapping/ resource visibility.
+
+    Composes a name and a fetch strategy
+    """
+
+    def __init__(self, name, conf):
+        """
+        Args:
+            name (str): Name of resource to be fetched
+            conf (dict): Dictionary representing resource endpoint layout
+        """
+        self._name = name
+        fetcher = spack.fetch_strategy.URLFetchStrategy(
+            url=conf["endpoint"], checksum=conf["sha256"]
+        )
+        stage = spack.stage.Stage(fetcher, path=str(windows_resource_root()))
+        resource = spack.resource.Resource(name, fetcher, destination=stage.path, placement=None)
+        self.stage = spack.stage.ResourceStage(
+            fetcher, stage, resource, path=str(windows_resource_root()), keep=True
+        )
```

Comment:
IMO it would be more idiomatic for this to:

* Create a stage without specifying a path (thereby downloading into the stage's tmp dir)
* From in the context manager, copy into `windows_resource_root`
* Allow `keep=False`, so the stage is cleaned up

vs. treating the `windows_resource_root` as the Stage root. Does that seem agreeable?

Reply:
Yeah that makes a lot of sense, will refactor.